sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/HexSuite.scala

```diff
@@ -37,4 +37,24 @@ class HexSuite extends SparkFunSuite {
     assert(Hex.hex(Long.MinValue).toString === "8000000000000000")
     assert(Hex.hex(Long.MaxValue).toString === "7FFFFFFFFFFFFFFF")
   }
+
+  test("SPARK-48644: hex bytes values") {
+    assert(Hex.hex(Array[Byte]()).toString === "")
+    assert(Hex.hex(Array[Byte](0, 1, 15, 16, 17, Byte.MaxValue)).toString === "00010F10117F")
+    assert(Hex.hex(Array[Byte](-1, -15, -16, -17, Byte.MinValue)).toString === "FFF1F0EF80")
+  }
+
+  test("SPARK-48644: bytes length check in hex") {
+    val bytes = new Array[Byte](1073741824)
```

Comment:
I didn't add this as it's quite memory consuming

Reply:
Thank you for pointing out. Make sense.